core/server/master/src/main/java/alluxio/master/meta/MetaMasterConfigurationServiceHandler.java

```diff
@@ -54,13 +62,35 @@ public void getConfiguration(GetConfigurationPOptions options,
   public void setPathConfiguration(SetPathConfigurationPRequest request,
       StreamObserver<SetPathConfigurationPResponse> responseObserver) {
     String path = request.getPath();
-    String key = request.getKey();
-    String value = request.getValue();
+    Map<String, String> properties = request.getPropertiesMap();
 
     RpcUtils.call(LOG, (RpcUtils.RpcCallableThrowsIOException<SetPathConfigurationPResponse>) () ->
     {
-      mMetaMaster.setPathConfiguration(path, PropertyKey.fromString(key), value);
+      Map<PropertyKey, String> props = new HashMap<>();
+      properties.forEach((key, value) -> props.put(PropertyKey.fromString(key), value));
+      mMetaMaster.setPathConfiguration(path, props);
       return SetPathConfigurationPResponse.getDefaultInstance();
     }, "setPathConfiguration", "request=%s", responseObserver, request);
   }
+
+  @Override
+  public void removePathConfiguration(RemovePathConfigurationPRequest request,
+      StreamObserver<RemovePathConfigurationPResponse> responseObserver) {
+    String path = request.getPath();
+    List<String> keys = request.getKeysList();
```

Comment:
Should we check `hasKeysList`? Default is empty but I think this is more explicit

Reply:
there is no such method

Reply:
Oh that's because the field is not optional, please use optional and not any required fields.

Reply:
it's a `repeated` field, cannot add `optional`